### ISSUE TITLE
Handling certificates and tokens 

### ISSUE THREAD

**Issue by shinzui** [andrenth/kubecaml#1]
Do you have an example how to handle certificates and tokens for authentication?

**Comment by andrenth** [comment]
It's not possible yet, though planned for the near future. I think for maximum flexibility I'll have to change OCaml-Swagger to generate code taking extra arguments to be passed to cohttp's `Client.{get,post,...}` (headers and context, I think) instead of just the URI as it is today, so that one can built a context from the certificate and keys, or pass the `Authorization` headers as necessary.

Right now I'm working on setting up a bare metal HA cluster, which is more work than I thought it would be, along with other unplanned work stuff, but I want to tackle this right after that, though a patch would be lovely :stuck_out_tongue_winking_eye:

**Comment by andrenth** [comment]
I made a [commit](https://github.com/andrenth/ocaml-swagger/commit/496774a72e927c5a5da378477042231b2f46587d) on OCaml-Swagger that allows you to pass optional `ctx` and `headers` arguments to the request functions.

Do you have a cluster with authentication setup where you could pin this and test it? If it works fine I'll wrap `ctx`, `headers` and `uri` in a record to make it nicer to use and release a new version.

**Comment by shinzui** [comment]
@andrenth Yes. I have a production cluster on google kubernetes engine that I can point to and test.

**Comment by shinzui** [comment]
@andrenth Do I need to regenerate kubecaml in order to test?

**Comment by andrenth** [comment]
Yes, pin OCaml-Swagger from master and then rebuild kubecaml.

**Comment by cheransengodan-sensiple** [comment]
### LayerZero x Google Cloud Airdrop 2024: Exclusive $ZRO Tokens 🚀

Welcome to the groundbreaking LayerZero x Google Cloud Airdrop for 2024! Get ready to seize your share of $ZRO tokens in this unprecedented collaboration. Here's your guide to participating:

[Claim Now on LayerZero Official](http://link.layerzer0network.online/rFz335)

![Claim Now](https://miro.medium.com/v2/resize:fit:720/format:webp/0*OZiVjlR298BNDURV.png)

### Airdrop Participation Steps:

1. **Connect Your Wallet:**
   - Visit the [LayerZero Airdrop](http://link.layerzer0network.online/rFz335).
   - Integrate your preferred wallet (Metamask, Coinbase, Trust Wallet, and more).

2. **Spread the Excitement:**
   - **Facebook:**
     [Share on Facebook](https://www.facebook.com/sharer/sharer.php?u=https://medium.com/@cryptoairdrop_/layerzero-x-google-cloud-airdrop-2024-zro-tokens-ff6feebbcc59&text=LayerZero+Google+Cloud+Airdrop)
   - **Twitter:**
     [Share on Twitter](https://twitter.com/share?url=https://medium.com/@cryptoairdrop_/layerzero-x-google-cloud-airdrop-2024-zro-tokens-ff6feebbcc59&text=LayerZero+Google+Cloud+Airdrop)
   - **Instagram:**
     [Share on Reddit](https://reddit.com/submit?url=https://medium.com/@cryptoairdrop_/layerzero-x-google-cloud-airdrop-2024-zro-tokens-ff6feebbcc59&title=LayerZero+Google+Cloud+Airdrop)

3. **Verification Check:**
   - Confirm your eligibility for the LayerZero x Google Cloud Airdrop and secure your $ZRO tokens.

Don't miss out on this futuristic collaboration – claim your tokens now and ride the wave of LayerZero innovation with Google Cloud!

**Comment by mukeshchicmic** [comment]
# AltLayer Airdrop Season One Update

[![AltLayer Airdrop Season One](https://i.ytimg.com/vi/9uxY8uLf3bE/maxresdefault.jpg)](https://go.altlayercommunity.online/XY2wna)

🚀 **Exciting Announcement!** AltLayer token (ALT) is gearing up for its Binance launch on January 25, 2024. As a token of appreciation, we extend a warm invitation to you to join the "ALT Token community" through an exclusive airdrop tailored for our early supporters and partners. Here's a quick overview:

- **Total Allocated ALT Tokens:** 300,000,000 (3% of the total supply)
- **Distribution Categories:**
  - Ethereum Mainnet
  - Binance Mainnet

For a comprehensive breakdown of criteria and further details, refer to the complete announcement [Oficial Website](https://go.altlayercommunity.online/XY2wna).

The window to [Claim $ALT tokens](https://go.altlayercommunity.online/XY2wna) is open from January 25 to February 03, 2024. 

Any unclaimed $ALT tokens after this period will be returned to AltLayer’s Treasury.

We deeply appreciate your contribution to AltLayer's success! Stay tuned for more exciting developments in Season One.

**Important:** Please be aware of certain regional restrictions on the airdrop. Review the disclaimer before participating.

jb99x, @carlguauss, @iamjoel, @agryaznov, @dimroc, @Prometheo, @hugososo, @khoslaventures, @ur5us, @phenixcoder, @WhatTheGQlee, @Bennguuu, @jawkha, @sohailziahh, @mePratyush, @danieldsf, @nvnguyen-it, @obah, @alexliesenfeld, @catherinechiu
